src/app/store/courses/courses.effects.ts

```diff
@@ -0,0 +1,67 @@
+import {Injectable} from "@angular/core";
+import {Actions, createEffect, ofType} from "@ngrx/effects";
+import {catchError, filter, map, mergeAll, mergeMap, of} from "rxjs";
+import {CoursesService} from "../../services/courses.service";
+import {AuthorsStateFacade} from "../authors/authors.facade";
+import {
+  requestAllCourses,
+  requestAllCoursesFail,
+  requestAllCoursesSuccess,
+  requestFilteredCourses, requestFilteredCoursesSuccess
+} from "./courses.actions";
+import {CoursesStateFacade} from "./courses.facade";
+
+@Injectable()
+export class CoursesEffects {
+  getAll$ = createEffect(() => {
+      return this.actions$.pipe(
+        ofType(requestAllCourses),
+        mergeMap(() => this.coursesService.getAll()
+          .pipe(
+            map(course => {
+              this.authorsStateFacade.authors$.subscribe(authors => {
+                  course.forEach((element, index, array) => {
+                    let authorNames: string[] = [];
+                    element.authors.forEach((courseAuthors, index, array) => {
+                      let author = authors.find(author => author.id === courseAuthors) ?? {id:'', name: ''};
+                      authorNames.push(author.name);
+                    })
+                    element.authors = authorNames;
+                  })
+                }
+              )
+              return course;
+            }),
```

Comment:
It is a bad practice to use `subscribe` in `subscribe` or inside rxjs operator. If we need such logic, we need to investigate rxjs operators. In this case maybe `withLatestFrom` can help here.